curve25519-dalek/src/backend/serial/u64/field_lemmas/limbs_to_bytes_lemmas.rs

```diff
@@ -0,0 +1,2396 @@
+#![allow(unused)]
+use vstd::arithmetic::div_mod::*;
+use vstd::arithmetic::mul::*;
+use vstd::arithmetic::power2::*;
+use vstd::bits::*;
+use vstd::prelude::*;
+
+use super::super::common_verus::bit_lemmas::*;
+use super::super::common_verus::div_mod_lemmas::*;
+use super::super::common_verus::mul_lemmas::*;
+use super::super::common_verus::pow_lemmas::*;
+use super::super::common_verus::shift_lemmas::*;
+use super::field_core::*;
+use super::load8_lemmas::*;
+use super::pow2_51_lemmas::*;
+
+verus! {
+
+// ============================================================================
+// LEMMA 3: Packing 51-bit limbs into 8-bit bytes
+// ============================================================================
+/// Predicate: bytes are packed from limbs according to the to_bytes algorithm
+/// This captures the byte-packing relationship used in FieldElement51::to_bytes (lines 380-410)
+pub open spec fn bytes_match_limbs_packing(limbs: [u64; 5], bytes: [u8; 32]) -> bool {
+    bytes[0] == limbs[0] as u8 && bytes[1] == (limbs[0] >> 8) as u8 && bytes[2] == (limbs[0]
+        >> 16) as u8 && bytes[3] == (limbs[0] >> 24) as u8 && bytes[4] == (limbs[0] >> 32) as u8
+        && bytes[5] == (limbs[0] >> 40) as u8 && bytes[6] == ((limbs[0] >> 48) | (limbs[1]
+        << 3)) as u8 && bytes[7] == (limbs[1] >> 5) as u8 && bytes[8] == (limbs[1] >> 13) as u8
+        && bytes[9] == (limbs[1] >> 21) as u8 && bytes[10] == (limbs[1] >> 29) as u8 && bytes[11]
+        == (limbs[1] >> 37) as u8 && bytes[12] == ((limbs[1] >> 45) | (limbs[2] << 6)) as u8
+        && bytes[13] == (limbs[2] >> 2) as u8 && bytes[14] == (limbs[2] >> 10) as u8 && bytes[15]
+        == (limbs[2] >> 18) as u8 && bytes[16] == (limbs[2] >> 26) as u8 && bytes[17] == (limbs[2]
+        >> 34) as u8 && bytes[18] == (limbs[2] >> 42) as u8 && bytes[19] == ((limbs[2] >> 50) | (
+    limbs[3] << 1)) as u8 && bytes[20] == (limbs[3] >> 7) as u8 && bytes[21] == (limbs[3]
+        >> 15) as u8 && bytes[22] == (limbs[3] >> 23) as u8 && bytes[23] == (limbs[3] >> 31) as u8
+        && bytes[24] == (limbs[3] >> 39) as u8 && bytes[25] == ((limbs[3] >> 47) | (limbs[4]
+        << 4)) as u8 && bytes[26] == (limbs[4] >> 4) as u8 && bytes[27] == (limbs[4] >> 12) as u8
+        && bytes[28] == (limbs[4] >> 20) as u8 && bytes[29] == (limbs[4] >> 28) as u8 && bytes[30]
+        == (limbs[4] >> 36) as u8 && bytes[31] == (limbs[4] >> 44) as u8
```

Comment:
This is one of my primary objections to using verusfmt, the formatted version is unreadable.